File: solutions/fortran/leap/1/leap.f90
module leap
  implicit none

contains

  logical function is_leap_year(year)
    integer, intent(in) :: year

    if (mod(year, 400) == 0) then
      is_leap_year = .true.
    else if (mod(year, 100) == 0) then
      is_leap_year = .false.
    else if (mod(year, 4) == 0) then
      is_leap_year = .true.
    else
      is_leap_year = .false.
    end if
  end function

end module

File: solutions/go/leap/1/leap.go
// Package leap provides a function to determine if a given year is a leap year.
package leap

// IsLeapYear returns true if the given year is a leap year according to the Gregorian calendar rules.
func IsLeapYear(year int) bool {
	return year%400 == 0 || (year%4 == 0 && year%100 != 0)
}